main/acle.md

```diff
@@ -7766,6 +7920,1419 @@ This is because the ACLE intrinsic calls do not imply a particular
+    s_callee(b);   // OK: streaming callee
+    sc_callee(i);  // OK: streaming callee
+    ls_callee(b);  // Undefined behavior: non-streaming callee
+  }
+
+  __attribute__((arm_streaming_compatible))
+  void sc_caller(void)
+  {
+    svint8_t i = ...;
+    svbool_t b = ...;
+
+    n_callee(i);   // Undefined behavior if sc_caller was called
+                   //   in streaming mode (but not otherwise)
+    s_callee(b);   // Undefined behavior if sc_caller was called
+                   //   in non-streaming mode (but not otherwise)
+    sc_callee(i);  // OK: keeps current streaming/non-streaming mode
+    ls_callee(b);  // Undefined behavior if sc_caller was called
+                   //   in streaming mode, but the call itself is
+                   //   OK otherwise. As noted above, the callee
+                   //   invokes undefined behavior internally
+  }
+```
+
+If the streaming and non-streaming vector lengths are equal then there
+is no undefined behavior in the code above.
+
+### `asm` restrictions related to streaming mode
+
+Some ACLE implementations might support the GNU “inline asm” extension.
+If so, the value of PSTATE.SM at the start of such inline asms is
+guaranteed to match the mode of the [abstract machine](#abstract-machine).
+For example, PSTATE.SM is guaranteed to be 1 at the start of inline asms
+in [streaming statements](#streaming-statement) and 0 at the start of
+inline asms in [non-streaming statements](#non-streaming-statement).
+
+All inline asms must preserve the value of PSTATE.SM.  Therefore:
+
+*   An asm entered in streaming mode must finish in streaming mode.
+
+*   An asm entered in non-streaming mode must finish in non-streaming mode.
+
+The behavior in other cases is undefined.
+
+An inline asm can temporarily switch mode internally, such as using an
+SMSTART/SMSTOP pair. However, such mode switches invalidate all Z and P
+register state. The asm's “clobber list” must mention any registers
+whose values might be changed by the asm.
+
+## ZA storage
+
+### Introduction to ZA storage
+
+SME provides an area of storage called ZA, of size SVL.B×SVL.B bytes.
+It also provides a processor state bit called PSTATE.ZA to control
+whether ZA is enabled.
+
+<!-- Do not remove the following `span`, they are needed to create an
+anchor that can be referred via an internal hyperlink to the paragraph
+following it. --><span id="za-state"></span>
+
+In C and C++ code, access to ZA is controlled at function granularity:
+a function either uses ZA or it does not. Another way to say this is
+that a function either “has ZA state” or it does not.
+
+If a function does have ZA state, the function can either share that ZA
+state with the function's caller or create new ZA state “from scratch”.
+In the latter case, it is the compiler's responsibility to free up ZA
+so that the function can use it; see the description of the lazy saving
+scheme in [[AAPCS64](#AAPCS64) for details about how the compiler
```

Comment:
```suggestion
scheme in [AAPCS64](#AAPCS64) for details about how the compiler
```

Reply:
Thanks for the catch.  I was going for `[[]]` (to be printed as `[AAPCS64]`) since that seemed to be the style preferred elsewhere for external references.

Reply:
@pthariensflame  - thank you for the comments. Are you happy for me to add you as a contributor to the [list of all-contributors](https://github.com/ARM-software/acle/#contributors-) for your review?